### ISSUE TITLE
opam init -c 4.07.0 is now ambiguous

### ISSUE THREAD

**Issue by pirbo** [ocaml/opam-repository#14075]
Sorry if this is the wrong channel.

Since the merge of #14065 
Probably because of the diff
```
diff --git a/packages/ocaml/ocaml.4.07.0/opam b/packages/ocaml/ocaml.4.07.0/opam
index 42d06f99ac..7641165207 100644
--- a/packages/ocaml/ocaml.4.07.0/opam
+++ b/packages/ocaml/ocaml.4.07.0/opam
@@ -27,3 +27,4 @@ authors: [
   "Didier Rémy"
   "Jérôme Vouillon"
 ]
+flags: compiler
```

`opam init -a --compiler=4.07.0` returns `[ERROR] Compiler selection '4.07.0' is ambiguous. matching packages: { ocaml.4.07.0, ocaml-base-compiler.4.07.0 }`

Is that the expected behaviour? How should I select my compiler now?

**Comment by mseri** [comment]
ping @dra27

**Comment by pirbo** [comment]
For the record, this has just been "Fixed" by #14079 . If this is indeed a proper fix. Please close this issue...
